MessageCorrelationTest.shouldCorrelateMessageAgainAfterRejection

**Summary**

> Try to answer the following as best as possible

- How often does the test fail? 4 times in the last 7 days
- Does it block your work? no
- Do we suspect that it is a real failure? no

**Failures**

> Outline known failure cases, e.g. a failed assertion and its stacktrace obtained from Jenkins

<details><summary>Example assertion failure</summary>
<pre>
java.lang.AssertionError: 
Expecting actual:
  [(EVENT, ELEMENT_ACTIVATING),
    (EVENT, ELEMENT_ACTIVATED),
    (COMMAND, ACTIVATE_ELEMENT),
    (EVENT, ELEMENT_ACTIVATING),
    (EVENT, ELEMENT_ACTIVATED),
    (COMMAND, COMPLETE_ELEMENT),
    (EVENT, ELEMENT_COMPLETING),
    (EVENT, ELEMENT_COMPLETED),
    (EVENT, SEQUENCE_FLOW_TAKEN),
    (COMMAND, ACTIVATE_ELEMENT),
    (EVENT, ELEMENT_ACTIVATING),
    (EVENT, CREATING),
    (EVENT, CREATING),
    (EVENT, ELEMENT_ACTIVATED),
    (COMMAND, CREATE),
    (COMMAND, CREATE),
    (EVENT, CREATED),
    (EVENT, CORRELATING),
    (COMMAND, CORRELATE),
    (EVENT, CREATED),
    (EVENT, CORRELATING),
    (COMMAND, CORRELATE),
    (EVENT, CORRELATED),
    (EVENT, TRIGGERING),
    (COMMAND, COMPLETE_ELEMENT),
    (COMMAND, CORRELATE),
    (COMMAND, REJECT),
    (COMMAND_REJECTION, CORRELATE),
    (EVENT, ELEMENT_COMPLETING),
    (EVENT, DELETING),
    (EVENT, ELEMENT_COMPLETED),
    (EVENT, TRIGGERED),
    (EVENT, ELEMENT_ACTIVATING),
    (EVENT, ELEMENT_ACTIVATED),
    (COMMAND, COMPLETE_ELEMENT),
    (COMMAND, DELETE),
    (EVENT, CORRELATED),
    (EVENT, REJECTED),
    (EVENT, ELEMENT_COMPLETING),
    (EVENT, ELEMENT_COMPLETED),
    (EVENT, SEQUENCE_FLOW_TAKEN),
    (COMMAND, ACTIVATE_ELEMENT),
    (EVENT, DELETED),
    (COMMAND, DELETE),
    (EVENT, ELEMENT_ACTIVATING),
    (EVENT, CREATING),
    (EVENT, ELEMENT_ACTIVATED),
    (COMMAND, CREATE),
    (EVENT, DELETED),
    (EVENT, CREATED),
    (EVENT, CORRELATING),
    (COMMAND, CORRELATE),
    (EVENT, CORRELATED),
    (EVENT, TRIGGERING),
    (COMMAND, COMPLETE_ELEMENT),
    (EVENT, ELEMENT_COMPLETING),
    (EVENT, ELEMENT_COMPLETED),
    (EVENT, SEQUENCE_FLOW_TAKEN),
    (COMMAND, ACTIVATE_ELEMENT),
    (COMMAND, CORRELATE),
    (EVENT, ELEMENT_ACTIVATING),
    (EVENT, ELEMENT_ACTIVATED),
    (EVENT, ELEMENT_COMPLETING),
    (EVENT, ELEMENT_COMPLETED),
    (EVENT, SEQUENCE_FLOW_TAKEN),
    (COMMAND, ACTIVATE_ELEMENT),
    (EVENT, CORRELATED),
    (EVENT, ELEMENT_ACTIVATING),
    (EVENT, ELEMENT_ACTIVATED),
    (EVENT, ELEMENT_COMPLETING),
    (EVENT, ELEMENT_COMPLETED),
    (COMMAND, COMPLETE_ELEMENT),
    (EVENT, ELEMENT_COMPLETING),
    (EVENT, ELEMENT_COMPLETED)]
to contain subsequence:
  [(COMMAND_REJECTION, CORRELATE),
    (COMMAND, REJECT),
    (EVENT, REJECTED),
    (COMMAND, CORRELATE)]
	at io.camunda.zeebe.engine.processing.message.MessageCorrelationTest.shouldCorrelateMessageAgainAfterRejection(MessageCorrelationTest.java:920)
</pre>
</details>

**Hypotheses**

> List any hypotheses if you have one; can be omitted

The order of the expected records is not stable. 

**Logs**

> If possible, provide more context here, e.g. standard output logs, link to build, etc.

<details><summary>Logs</summary>
<pre>
09:38:43.068 [Broker-0-LogStream-1] DEBUG io.camunda.zeebe.logstreams - Configured log appender back pressure at partition 1 as AppenderVegasCfg{initialLimit=1024, maxConcurrency=32768, alphaLimit=0.7, betaLimit=0.95}. Window limiting is disabled
09:38:43.201 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Recovering state of partition 1 from snapshot
09:38:43.203 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@37a8cc45)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@396f24c0, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@3d50a735, configuration: Configuration(false)]
09:38:43.204 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.logstreams - Recovered state of partition 1 from snapshot at position -1
09:38:43.204 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@78522efb)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@9690f50, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@49607399, configuration: Configuration(false)]
09:38:43.205 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@71b10c8f)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@187dd1b4, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@59af5e64, configuration: Configuration(false)]
09:38:43.206 [Broker-0-StreamProcessor-1] INFO  org.camunda.feel.FeelEngine - Engine created. [value-mapper: CompositeValueMapper(List(io.camunda.zeebe.el.impl.feel.MessagePackValueMapper@13991273)), function-provider: io.camunda.zeebe.el.impl.feel.FeelFunctionProvider@6ab5a1fb, clock: io.camunda.zeebe.el.impl.ZeebeFeelEngineClock@1c332d68, configuration: Configuration(false)]
09:38:43.245 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.processor - Processor starts replay of events. [snapshot-position: -1, replay-mode: PROCESSING]
09:38:43.245 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.processor - Processor finished replay at event position -1
09:38:43.245 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Starting processing of migration tasks (use LogLevel.DEBUG for more details) ... 
09:38:43.246 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.engine.state.migration - Found 2 migration tasks: ProcessMessageSubscriptionSentTimeMigration, MessageSubscriptionSentTimeMigration
09:38:43.246 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Skipping ProcessMessageSubscriptionSentTimeMigration migration (1/2).  It was determined it does not need to run right now.
09:38:43.246 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Skipping MessageSubscriptionSentTimeMigration migration (2/2).  It was determined it does not need to run right now.
09:38:43.246 [Broker-0-StreamProcessor-1] INFO  io.camunda.zeebe.engine.state.migration - Completed processing of migration tasks (use LogLevel.DEBUG for more details) ... 
09:38:43.247 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.engine.state.migration - Executed 0 migration tasks: 
09:38:43.478 [] INFO  io.camunda.zeebe.test - Test failed, following records were exported:
09:38:43.647 [] INFO  io.camunda.zeebe.test - Compact log representation:
--------
	['C'ommand/'E'event/'R'ejection] [valueType] [intent] - #[position]->#[source record position]  P[partitionId]K[key] - [summary of value]
	P9K999 - key; #999 - record position; "ID" element/process id; @"elementid"/[P9K999] - element with ID and key
	Keys are decomposed into partition id and per partition key (e.g. 2251799813685253 -> P1K005). If single partition, the partition is omitted.
	Long IDs are shortened (e.g. 'startEvent_5d56488e-0570-416c-ba2d-36d2a3acea78' -> 'star..acea78'
--------
C MSG                PUBLISH        - #01-> -1  -1 - "a" correlationKey: 123 (no vars)
E MSG                PUBLISHED      - #02->#01 K01 - "a" correlationKey: 123 (no vars)
C MSG                PUBLISH        - #03-> -1  -1 - "b" correlationKey: 123 (no vars)
E MSG                PUBLISHED      - #04->#03 K02 - "b" correlationKey: 123 (no vars)
C DEPLOYMENT         CREATE         - #05-> -1  -1 - 
E PROC               CREATED        - #06->#05 K03 - process.xml -> "process" (version:1)
E DEPLOYMENT         CREATED        - #07->#05 K04 - process.xml
E DEPLOYMENT         FULLY_DISTR    - #08->#05 K04 - 
C PROC_INST_CREATION CREATE         - #09-> -1  -1 - new <process "process"> with variables: {key=123}
E VAR                CREATED        - #10->#09 K06 - key->"123" in <process [K05]>
C PROC_INST          ACTIVATE       - #11->#09 K05 - PROCESS "process" in <process "process"[K05]>
E PROC_INST_CREATION CREATED        - #12->#09 K07 - new <process "process"> with variables: {key=123}
E PROC_INST          ACTIVATING     - #13->#11 K05 - PROCESS "process" in <process "process"[K05]>
E PROC_INST          ACTIVATED      - #14->#11 K05 - PROCESS "process" in <process "process"[K05]>
C PROC_INST          ACTIVATE       - #15->#11  -1 - START_EVENT "startEv..cddb2e6" in <process "process"[K05]>
E PROC_INST          ACTIVATING     - #16->#15 K08 - START_EVENT "startEv..cddb2e6" in <process "process"[K05]>
E PROC_INST          ACTIVATED      - #17->#15 K08 - START_EVENT "startEv..cddb2e6" in <process "process"[K05]>
C PROC_INST          COMPLETE       - #18->#15 K08 - START_EVENT "startEv..cddb2e6" in <process "process"[K05]>
E PROC_INST          COMPLETING     - #19->#18 K08 - START_EVENT "startEv..cddb2e6" in <process "process"[K05]>
E PROC_INST          COMPLETED      - #20->#18 K08 - START_EVENT "startEv..cddb2e6" in <process "process"[K05]>
E PROC_INST          SEQ_FLOW_TAKEN - #21->#18 K09 - SEQUENCE_FLOW "sequenc..492b330" in <process "process"[K05]>
C PROC_INST          ACTIVATE       - #22->#18 K10 - EVENT_BASED_GATEWAY "split" in <process "process"[K05]>
E PROC_INST          ACTIVATING     - #23->#22 K10 - EVENT_BASED_GATEWAY "split" in <process "process"[K05]>
E PROC_MSG_SUB       CREATING       - #24->#22 K11 - "b" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E PROC_MSG_SUB       CREATING       - #25->#22 K12 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E PROC_INST          ACTIVATED      - #26->#22 K10 - EVENT_BASED_GATEWAY "split" in <process "process"[K05]>
C MSG_SUB            CREATE         - #27-> -1  -1 - "b" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
C MSG_SUB            CREATE         - #28-> -1  -1 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E MSG_SUB            CREATED        - #29->#27 K13 - "b" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E MSG_SUB            CORRELATING    - #30->#27 K13 - "b" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
C PROC_MSG_SUB       CORRELATE      - #31-> -1  -1 - "b" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E MSG_SUB            CREATED        - #32->#28 K14 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E MSG_SUB            CORRELATING    - #33->#28 K14 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
C PROC_MSG_SUB       CORRELATE      - #34-> -1  -1 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E PROC_MSG_SUB       CORRELATED     - #35->#31 K11 - "b" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E PROC_EVNT          TRIGGERING     - #36->#31 K15 -  @"element-b"[K10] in <process K03[K05]> (no vars)
C PROC_INST          COMPLETE       - #37->#31 K10 - EVENT_BASED_GATEWAY "split" in <process "process"[K05]>
C MSG_SUB            CORRELATE      - #38-> -1  -1 - "b" (inter.) @[K10] in <process "process"[K05]> (no vars)
C MSG_SUB            REJECT         - #39-> -1  -1 - "a" correlationKey: 123 @[ -1] in <process "process"[K05]> (no vars)
R PROC_MSG_SUB       CORRELATE      - #40->#34  -1 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars) !INVALID_STATE (Expected to correlate a process message subscription with element key '2251799813685258' and message name 'a', but the subscription is not active anymore)
E PROC_INST          COMPLETING     - #41->#37 K10 - EVENT_BASED_GATEWAY "split" in <process "process"[K05]>
E PROC_MSG_SUB       DELETING       - #42->#37 K12 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E PROC_INST          COMPLETED      - #43->#37 K10 - EVENT_BASED_GATEWAY "split" in <process "process"[K05]>
E PROC_EVNT          TRIGGERED      - #44->#37 K15 -  @"element-b"[K10] in <process K03[K05]> (no vars)
E PROC_INST          ACTIVATING     - #45->#37 K16 - INTERMEDIATE_CATCH_EVENT "element-b" in <process "process"[K05]>
E PROC_INST          ACTIVATED      - #46->#37 K16 - INTERMEDIATE_CATCH_EVENT "element-b" in <process "process"[K05]>
C PROC_INST          COMPLETE       - #47->#37 K16 - INTERMEDIATE_CATCH_EVENT "element-b" in <process "process"[K05]>
C MSG_SUB            DELETE         - #48-> -1  -1 - "a" (inter.) @[K10] in <process ?[K05]> (no vars)
E MSG_SUB            CORRELATED     - #49->#38 K13 - "b" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E MSG_SUB            REJECTED       - #50->#39  -1 - "a" correlationKey: 123 @[ -1] in <process "process"[K05]> (no vars)
E PROC_INST          COMPLETING     - #51->#47 K16 - INTERMEDIATE_CATCH_EVENT "element-b" in <process "process"[K05]>
E PROC_INST          COMPLETED      - #52->#47 K16 - INTERMEDIATE_CATCH_EVENT "element-b" in <process "process"[K05]>
E PROC_INST          SEQ_FLOW_TAKEN - #53->#47 K17 - SEQUENCE_FLOW "sequenc..902f7eb" in <process "process"[K05]>
C PROC_INST          ACTIVATE       - #54->#47 K18 - INTERMEDIATE_CATCH_EVENT "element-ba" in <process "process"[K05]>
E MSG_SUB            DELETED        - #55->#48 K14 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
C PROC_MSG_SUB       DELETE         - #56-> -1  -1 - "a" (inter.) @[K10] in <process ?[K05]> (no vars)
E PROC_INST          ACTIVATING     - #57->#54 K18 - INTERMEDIATE_CATCH_EVENT "element-ba" in <process "process"[K05]>
E PROC_MSG_SUB       CREATING       - #58->#54 K19 - "a" (inter.) correlationKey: 123 @[K18] in <process "process"[K05]> (no vars)
E PROC_INST          ACTIVATED      - #59->#54 K18 - INTERMEDIATE_CATCH_EVENT "element-ba" in <process "process"[K05]>
C MSG_SUB            CREATE         - #60-> -1  -1 - "a" (inter.) correlationKey: 123 @[K18] in <process "process"[K05]> (no vars)
E PROC_MSG_SUB       DELETED        - #61->#56 K12 - "a" (inter.) correlationKey: 123 @[K10] in <process "process"[K05]> (no vars)
E MSG_SUB            CREATED        - #62->#60 K20 - "a" (inter.) correlationKey: 123 @[K18] in <process "process"[K05]> (no vars)
E MSG_SUB            CORRELATING    - #63->#60 K20 - "a" (inter.) correlationKey: 123 @[K18] in <process "process"[K05]> (no vars)
C PROC_MSG_SUB       CORRELATE      - #64-> -1  -1 - "a" (inter.) correlationKey: 123 @[K18] in <process "process"[K05]> (no vars)
E PROC_MSG_SUB       CORRELATED     - #65->#64 K19 - "a" (inter.) correlationKey: 123 @[K18] in <process "process"[K05]> (no vars)
E PROC_EVNT          TRIGGERING     - #66->#64 K21 -  @"element-ba"[K18] in <process K03[K05]> (no vars)
C PROC_INST          COMPLETE       - #67->#64 K18 - INTERMEDIATE_CATCH_EVENT "element-ba" in <process "process"[K05]>
E PROC_INST          COMPLETING     - #68->#67 K18 - INTERMEDIATE_CATCH_EVENT "element-ba" in <process "process"[K05]>
E PROC_INST          COMPLETED      - #69->#67 K18 - INTERMEDIATE_CATCH_EVENT "element-ba" in <process "process"[K05]>
E PROC_INST          SEQ_FLOW_TAKEN - #70->#67 K22 - SEQUENCE_FLOW "sequenc..3f4421a" in <process "process"[K05]>
C PROC_INST          ACTIVATE       - #71->#67 K23 - EXCLUSIVE_GATEWAY "merge" in <process "process"[K05]>
C MSG_SUB            CORRELATE      - #72-> -1  -1 - "a" (inter.) @[K18] in <process "process"[K05]> (no vars)
E PROC_INST          ACTIVATING     - #73->#71 K23 - EXCLUSIVE_GATEWAY "merge" in <process "process"[K05]>
E PROC_INST          ACTIVATED      - #74->#71 K23 - EXCLUSIVE_GATEWAY "merge" in <process "process"[K05]>
E PROC_INST          COMPLETING     - #75->#71 K23 - EXCLUSIVE_GATEWAY "merge" in <process "process"[K05]>
E PROC_INST          COMPLETED      - #76->#71 K23 - EXCLUSIVE_GATEWAY "merge" in <process "process"[K05]>
E PROC_INST          SEQ_FLOW_TAKEN - #77->#71 K24 - SEQUENCE_FLOW "sequenc..359c255" in <process "process"[K05]>
C PROC_INST          ACTIVATE       - #78->#71 K25 - END_EVENT "endEven..63fb8a1" in <process "process"[K05]>
E MSG_SUB            CORRELATED     - #79->#72 K20 - "a" (inter.) correlationKey: 123 @[K18] in <process "process"[K05]> (no vars)
E PROC_INST          ACTIVATING     - #80->#78 K25 - END_EVENT "endEven..63fb8a1" in <process "process"[K05]>
E PROC_INST          ACTIVATED      - #81->#78 K25 - END_EVENT "endEven..63fb8a1" in <process "process"[K05]>
E PROC_INST          COMPLETING     - #82->#78 K25 - END_EVENT "endEven..63fb8a1" in <process "process"[K05]>
E PROC_INST          COMPLETED      - #83->#78 K25 - END_EVENT "endEven..63fb8a1" in <process "process"[K05]>
C PROC_INST          COMPLETE       - #84->#78 K05 - PROCESS "process" in <process "process"[K05]>
E PROC_INST          COMPLETING     - #85->#84 K05 - PROCESS "process" in <process "process"[K05]>
E PROC_INST          COMPLETED      - #86->#84 K05 - PROCESS "process" in <process "process"[K05]>
-------------- Deployed Processes ----------------------
process.xml -> "process" (version:1)[K03] ------
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<definitions xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI" xmlns:dc="http://www.omg.org/spec/DD/20100524/DC" xmlns:di="http://www.omg.org/spec/DD/20100524/DI" xmlns:ns0="http://camunda.org/schema/zeebe/1.0" id="definitions_adc0eef2-28ee-4860-9f9b-cb3e128b7133" targetNamespace="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL">
  <process id="process" isExecutable="true">
    <startEvent id="startEvent_7a5ed038-4e39-44f6-b63a-f5398cddb2e6">
      <outgoing>sequenceFlow_32a1fae8-fd92-4040-8402-64643492b330</outgoing>
    </startEvent>
    <eventBasedGateway id="split" name="split">
      <incoming>sequenceFlow_32a1fae8-fd92-4040-8402-64643492b330</incoming>
      <outgoing>sequenceFlow_d41d45f5-a6e0-4277-831a-84bad333d862</outgoing>
      <outgoing>sequenceFlow_8e0b0952-df8e-48ba-959b-148052c152a9</outgoing>
    </eventBasedGateway>
    <sequenceFlow id="sequenceFlow_32a1fae8-fd92-4040-8402-64643492b330" sourceRef="startEvent_7a5ed038-4e39-44f6-b63a-f5398cddb2e6" targetRef="split"/>
    <intermediateCatchEvent id="element-a" name="element-a">
      <incoming>sequenceFlow_d41d45f5-a6e0-4277-831a-84bad333d862</incoming>
      <outgoing>sequenceFlow_0dce6569-92e1-45bb-b1e9-860c84ad07f3</outgoing>
      <messageEventDefinition id="messageEventDefinition_30431f79-413b-4e98-a39a-3ccee3ce1962" messageRef="message_609e1b3c-ecbe-40a4-b10e-6325449a4924"/>
    </intermediateCatchEvent>
    <sequenceFlow id="sequenceFlow_d41d45f5-a6e0-4277-831a-84bad333d862" sourceRef="split" targetRef="element-a"/>
    <intermediateCatchEvent id="element-ab" name="element-ab">
      <incoming>sequenceFlow_0dce6569-92e1-45bb-b1e9-860c84ad07f3</incoming>
      <outgoing>sequenceFlow_837ac5fe-e87e-42b0-81b3-4be4b27af1bd</outgoing>
      <messageEventDefinition id="messageEventDefinition_0988bc6d-79c5-46fa-aa09-ad2e835abcf0" messageRef="message_1d3895f3-9313-40b4-b913-418dd9620c80"/>
    </intermediateCatchEvent>
    <sequenceFlow id="sequenceFlow_0dce6569-92e1-45bb-b1e9-860c84ad07f3" sourceRef="element-a" targetRef="element-ab"/>
    <exclusiveGateway id="merge" name="merge">
      <incoming>sequenceFlow_837ac5fe-e87e-42b0-81b3-4be4b27af1bd</incoming>
      <incoming>sequenceFlow_7e790997-a4d4-4443-81e8-a3b4c3f4421a</incoming>
      <outgoing>sequenceFlow_2022f213-e14b-4207-9481-ba15c359c255</outgoing>
    </exclusiveGateway>
    <sequenceFlow id="sequenceFlow_837ac5fe-e87e-42b0-81b3-4be4b27af1bd" sourceRef="element-ab" targetRef="merge"/>
    <endEvent id="endEvent_ff6d9d39-6aa2-45bf-81ca-9638363fb8a1">
      <incoming>sequenceFlow_2022f213-e14b-4207-9481-ba15c359c255</incoming>
    </endEvent>
    <sequenceFlow id="sequenceFlow_2022f213-e14b-4207-9481-ba15c359c255" sourceRef="merge" targetRef="endEvent_ff6d9d39-6aa2-45bf-81ca-9638363fb8a1"/>
    <intermediateCatchEvent id="element-b" name="element-b">
      <incoming>sequenceFlow_8e0b0952-df8e-48ba-959b-148052c152a9</incoming>
      <outgoing>sequenceFlow_c8d626e5-89c9-4e53-a044-a3c2b902f7eb</outgoing>
      <messageEventDefinition id="messageEventDefinition_9694b1ee-e335-4a30-8466-f10ca4367b80" messageRef="message_f199264d-58bf-4f4c-91c5-c0ce9ceb4f3f"/>
    </intermediateCatchEvent>
    <sequenceFlow id="sequenceFlow_8e0b0952-df8e-48ba-959b-148052c152a9" sourceRef="split" targetRef="element-b"/>
    <intermediateCatchEvent id="element-ba" name="element-ba">
      <incoming>sequenceFlow_c8d626e5-89c9-4e53-a044-a3c2b902f7eb</incoming>
      <outgoing>sequenceFlow_7e790997-a4d4-4443-81e8-a3b4c3f4421a</outgoing>
      <messageEventDefinition id="messageEventDefinition_f70c0ba8-5e33-4e3d-9a99-660e0f6633d5" messageRef="message_f78b6df0-5ac7-4ca2-b4cf-b7b06761b8be"/>
    </intermediateCatchEvent>
    <sequenceFlow id="sequenceFlow_c8d626e5-89c9-4e53-a044-a3c2b902f7eb" sourceRef="element-b" targetRef="element-ba"/>
    <sequenceFlow id="sequenceFlow_7e790997-a4d4-4443-81e8-a3b4c3f4421a" sourceRef="element-ba" targetRef="merge"/>
  </process>
  <message id="message_609e1b3c-ecbe-40a4-b10e-6325449a4924" name="a">
    <extensionElements>
      <ns0:subscription correlationKey="=key"/>
    </extensionElements>
  </message>
  <message id="message_1d3895f3-9313-40b4-b913-418dd9620c80" name="b">
    <extensionElements>
      <ns0:subscription correlationKey="=key"/>
    </extensionElements>
  </message>
  <message id="message_f199264d-58bf-4f4c-91c5-c0ce9ceb4f3f" name="b">
    <extensionElements>
      <ns0:subscription correlationKey="=key"/>
    </extensionElements>
  </message>
  <message id="message_f78b6df0-5ac7-4ca2-b4cf-b7b06761b8be" name="a">
    <extensionElements>
      <ns0:subscription correlationKey="=key"/>
    </extensionElements>
  </message>
  <bpmndi:BPMNDiagram id="BPMNDiagram_01793873-f359-495b-baa3-7431c51f79f6">
    <bpmndi:BPMNPlane bpmnElement="process" id="BPMNPlane_1d22883c-57fe-4721-b706-5f439001daa6">
      <bpmndi:BPMNShape bpmnElement="startEvent_7a5ed038-4e39-44f6-b63a-f5398cddb2e6" id="BPMNShape_f573d054-d4d4-4902-af6a-7133cfc9f9c0">
        <dc:Bounds height="36.0" width="36.0" x="100.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNShape bpmnElement="split" id="BPMNShape_7bd267a6-65ce-4e88-9350-8ebcb5981b2a">
        <dc:Bounds height="50.0" width="50.0" x="186.0" y="93.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_32a1fae8-fd92-4040-8402-64643492b330" id="BPMNEdge_0de7e56f-8cd7-4957-8f68-38d3455a42ee">
        <di:waypoint x="136.0" y="118.0"/>
        <di:waypoint x="186.0" y="118.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="element-a" id="BPMNShape_25a55b1f-af82-4e81-af42-625a54bb70dc">
        <dc:Bounds height="36.0" width="36.0" x="286.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_d41d45f5-a6e0-4277-831a-84bad333d862" id="BPMNEdge_281db2ad-e12d-499e-9b47-2003cecef269">
        <di:waypoint x="236.0" y="118.0"/>
        <di:waypoint x="286.0" y="118.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="element-ab" id="BPMNShape_f3c66cbe-e1fc-4211-922a-521043e5fcc1">
        <dc:Bounds height="36.0" width="36.0" x="372.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_0dce6569-92e1-45bb-b1e9-860c84ad07f3" id="BPMNEdge_0c5459cf-3afd-4a95-8224-f3b06bb313c1">
        <di:waypoint x="322.0" y="118.0"/>
        <di:waypoint x="372.0" y="118.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="merge" id="BPMNShape_b5010a5f-9fa2-402f-bf75-10c97561593c" isMarkerVisible="true">
        <dc:Bounds height="50.0" width="50.0" x="458.0" y="93.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_837ac5fe-e87e-42b0-81b3-4be4b27af1bd" id="BPMNEdge_29bc4507-42c9-49dd-a071-5f5462d8e7a5">
        <di:waypoint x="408.0" y="118.0"/>
        <di:waypoint x="458.0" y="118.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="endEvent_ff6d9d39-6aa2-45bf-81ca-9638363fb8a1" id="BPMNShape_853a4db8-d669-4b69-99a2-c59ed35f43dc">
        <dc:Bounds height="36.0" width="36.0" x="558.0" y="100.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_2022f213-e14b-4207-9481-ba15c359c255" id="BPMNEdge_2a36a972-cce1-4401-80dc-6e8fbea7cd9d">
        <di:waypoint x="508.0" y="118.0"/>
        <di:waypoint x="558.0" y="118.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="element-b" id="BPMNShape_197bffb2-641a-4ede-ad17-ba7e6867230d">
        <dc:Bounds height="36.0" width="36.0" x="286.0" y="186.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_8e0b0952-df8e-48ba-959b-148052c152a9" id="BPMNEdge_b55fd863-7b17-495f-8c93-9b469885b008">
        <di:waypoint x="211.0" y="143.0"/>
        <di:waypoint x="211.0" y="204.0"/>
        <di:waypoint x="286.0" y="204.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNShape bpmnElement="element-ba" id="BPMNShape_df791082-d1f6-4417-af9c-a9f96af5fcc2">
        <dc:Bounds height="36.0" width="36.0" x="372.0" y="186.0"/>
      </bpmndi:BPMNShape>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_c8d626e5-89c9-4e53-a044-a3c2b902f7eb" id="BPMNEdge_0b73636e-863b-4115-9e85-b858f938fa5f">
        <di:waypoint x="322.0" y="204.0"/>
        <di:waypoint x="372.0" y="204.0"/>
      </bpmndi:BPMNEdge>
      <bpmndi:BPMNEdge bpmnElement="sequenceFlow_7e790997-a4d4-4443-81e8-a3b4c3f4421a" id="BPMNEdge_53744bd8-6d64-4bd2-a7e9-231e0fabe025">
        <di:waypoint x="408.0" y="204.0"/>
        <di:waypoint x="458.0" y="118.0"/>
      </bpmndi:BPMNEdge>
    </bpmndi:BPMNPlane>
  </bpmndi:BPMNDiagram>
</definitions>
--------------- Decomposed keys (for debugging) -----------------
 -1 <-> -1
K01 <-> 2251799813685249
K02 <-> 2251799813685250
K03 <-> 2251799813685251
K04 <-> 2251799813685252
K05 <-> 2251799813685253
K06 <-> 2251799813685254
K07 <-> 2251799813685255
K08 <-> 2251799813685256
K09 <-> 2251799813685257
K10 <-> 2251799813685258
K11 <-> 2251799813685259
K12 <-> 2251799813685260
K13 <-> 2251799813685261
K14 <-> 2251799813685262
K15 <-> 2251799813685263
K16 <-> 2251799813685264
K17 <-> 2251799813685265
K18 <-> 2251799813685266
K19 <-> 2251799813685267
K20 <-> 2251799813685268
K21 <-> 2251799813685269
K22 <-> 2251799813685270
K23 <-> 2251799813685271
K24 <-> 2251799813685272
K25 <-> 2251799813685273
09:38:43.651 [] DEBUG io.camunda.zeebe.util.buffer - Close stream processor
09:38:43.652 [Broker-0-StreamProcessor-1] DEBUG io.camunda.zeebe.logstreams - Closed stream processor controller Broker-0-StreamProcessor-1.
09:38:43.691 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - Close appender for log stream stream-1
09:38:43.692 [stream-1-write-buffer] DEBUG io.camunda.zeebe.dispatcher - Dispatcher closed
09:38:43.692 [Broker-0-LogStream-1] INFO  io.camunda.zeebe.logstreams - On closing logstream stream-1 close 2 readers
09:38:43.692 [] DEBUG io.camunda.zeebe.broker.test - Clean up test files on path /tmp/junit12022277910273450954
09:38:43.693 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers'
09:38:43.693 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors'
09:38:43.693 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-fs-workers': closed successfully
09:38:43.696 [] DEBUG io.camunda.zeebe.util.actor - Closing actor thread ground '-zb-actors': closed successfully
</pre>
</details>

--- a/engine/src/test/java/io/camunda/zeebe/engine/processing/message/MessageCorrelationTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/processing/message/MessageCorrelationTest.java
@@ -923,7 +923,6 @@ public final class MessageCorrelationTest {
         .extracting(Record::getRecordType, Record::getIntent)
         .containsSubsequence(
             tuple(RecordType.COMMAND_REJECTION, ProcessMessageSubscriptionIntent.CORRELATE),
-            tuple(RecordType.COMMAND, MessageSubscriptionIntent.REJECT),
             tuple(RecordType.EVENT, MessageSubscriptionIntent.REJECTED),
             tuple(RecordType.COMMAND, ProcessMessageSubscriptionIntent.CORRELATE));
 

